creator-node/src/routes/nodeSync.js

```diff
@@ -53,7 +58,15 @@ module.exports = function (app) {
         cnodeUserDictObj['files'] = []
 
         cnodeUsersDict[cnodeUser.cnodeUserUUID] = cnodeUserDictObj
-      })
+
+        // Rely on Redis cache to dedupe rehydrating files from cnode UUIDs that have been recently hydrated
+        const entry = await redisClient.get(`recently_rehydrated_wallet_${cnodeUser.walletPublicKey}`)
+        if (!entry) {
+          await redisClient.set(`recently_rehydrated_wallet_${cnodeUser.walletPublicKey}`, cnodeUser.cnodeUserUUID, 'EX', TTL)
+          cnodeUserUUIDsOfFilesToRehydrateArrayPromises.push(Promise.resolve(cnodeUser.cnodeUserUUID))
```

Comment:
not sure what this promise.resolve is for, same with the PRomise.all in the next line? this is just pushing the cnodeUserUUID onto an array?

Reply:
yes, this is to ensure that the array of uuids to rehydrate is fully propagated before we continue with the rest of the export and rehydrate logic. if left without awaiting all the returned `Promise.resolve(uuid)` objects in the array,  the rehydrate process might skip over some uuids